app/grandchallenge/core/utils/grand_challenge_forge.py

```diff
@@ -22,10 +46,41 @@ def get_forge_challenge_pack_context(challenge, phase_pks=None):
     grand-challenge-forge to generate a challenge pack.
     """
 
-    phases = challenge.phase_set.filter(
-        archive__isnull=False,
-        submission_kind=SubmissionKindChoices.ALGORITHM,
-    ).prefetch_related("archive", "algorithm_interfaces")
+    from grandchallenge.components.models import ComponentInterface
+
+    phases = (
+        challenge.phase_set.filter(
+            archive__isnull=False,
+            submission_kind=SubmissionKindChoices.ALGORITHM,
+        )
+        .order_by("slug")
+        .prefetch_related(
+            Prefetch(
+                "archive",
+                queryset=Archive.objects.order_by("created"),
+            ),
+            Prefetch(
+                "additional_evaluation_inputs",
+                queryset=ComponentInterface.objects.order_by("slug"),
```

Comment:
Why force ordering for all of these things, including the interfaces below? This changes the ordering from what users see in the UI.

Reply:
Consistent generation of templates: there is no innate ordering on the contained sockets. IMHO regenerating the template should be a repeatable action.

Reply:
They should be ordered by pk already https://github.com/comic/grand-challenge.org/blob/f2a784cb8ae1cb4575a767ceb219402267496e13/app/grandchallenge/components/models.py#L703

Reply:
Ah, not sure how I missed that! It's been there for over 4 years!

I do remember a different point about `Meta.ordering` long ago: explicit usage that requires ordering (or assumes it) should explicit require it. And having a 'central' ordering was generally lead to a lot of useless DB actions.

Ofc we now already have that ordering in the `CI.Meta`. So I think it is fine to not be explicit about it downstream since in practice we never remove central ordering? Would you agree?

Reply:
We should do ordering in one place which is in the models Meta method. That way we can ensure that the appropriate indices are set so that ordering is efficient as it is never free (gratis).

Of course there are cases where explicit ordering is needed, but there needs to be a good reason for it. In this case, I think we can go with the default ordering everywhere.

Reply:
Ah I see now why I originally did this: The `AlgorithmInterface` are not ordered by default. Should we, in this case, add a central `AlgorithmInterface.Meta.ordering` or here explicitly?

Moreover, the `"created"` would be my first to go for ordering but it seems those are not indiced by default. Considering all related DB queries will not really have a large number of final objects would it be fine to pick a non-indiced column?

Reply:
Yes that would be fine.